### DISCUSSION TITLE
Authentication Request goes unanswered with Pixel 5

### DISCUSSION THREAD

**Discussion by caseyonline** [srsran/srsRAN_4G#1431] [Q&A]
I’m consistently observing a successful Setup three way handshake when connecting my Pixel 5 to my srsran private network. However, the Authentication request (0x52) that follows on the DL to the UE goes unanswered with no indication of why in the logs nor a reject nor failure response (pcap attached)

I’ve confirmed that the IMSI, KI and OPC codes match between the SIM and the user_db.csv file (attached details for both).

I run epc first, then enb (attached logs for both)

Looking at android logs (adb logcat -b radio,events,system,main) doesn’t help me understand this impasse. Are there logs that might provide more detail on the UE side?

I’m using an mcc=001 and a mnc=01. Is that an issue? I’ve also tried mcc=999 with a similar outcome.

Setup
Ettus USRP B205
Pixel 5 (rooted) with Android 14
Ubuntu 22.04 hosting epc and enb services.

![Screenshot (2)](https://github.com/user-attachments/assets/fc3599ad-3f09-413a-9164-e7af9cd47076)

[user_db.csv.txt](https://github.com/user-attachments/files/17998769/user_db.csv.txt)
[rr.conf.txt](https://github.com/user-attachments/files/17998770/rr.conf.txt)
[epc.conf.txt](https://github.com/user-attachments/files/17998771/epc.conf.txt)
[enb.conf.txt](https://github.com/user-attachments/files/17998772/enb.conf.txt)
[epc.log.txt](https://github.com/user-attachments/files/17998773/epc.log.txt)
[enb.trace.txt](https://github.com/user-attachments/files/17998774/enb.trace.txt)
[enb_mac.pcap.txt](https://github.com/user-attachments/files/17998775/enb_mac.pcap.txt)

**Comment by caseyonline** [comment]
I would expect to see a Authentication failure if something goes wrong,
https://github.com/srsran/srsRAN_4G/blob/master/srsue/src/stack/upper/nas.cc#L1119
but I don't know if the OEM modem software on my pixel behaves the same

**Comment by caseyonline** [comment]
![lte_enb](https://github.com/user-attachments/assets/a7c5e393-35db-4251-905f-62235b70e8ee)

**Reply by andrepuschmann** [reply]
It seems the phone RLC ACKs the reception of the auth request but then never responds. Could you provide enb logs in PHY/MAC info mode to see if there is any further SR from the UE?

**Reply by caseyonline** [reply]
appreciate the response .. I've attached the enb logs for all levels at debug level.   Thanks

[enb.log.zip](https://github.com/user-attachments/files/18065222/enb.log.zip)

**Reply by andrepuschmann** [reply]
looks like the phone does not respond after the RLC ack - no idea why though

**Reply by caseyonline** [reply]
.. looks like I need to pursue getting more info from the logs on the Pixel 5 ..

**Comment by caseyonline** [comment]
I'm not having any luck getting detailed logs from the phone.   'adb logcat -b radio' doesn't help.  I can only speculate as to what's going on at the mobile,

1. The auth request message isn't recognized as valid and is thrown away
2. The RRC/NAS Layer 3 is in a state that doesn't expect the auth request and ignores it

I tried this on a Samsung Galaxy S9 phone (production) with similar results.   Both these phones have the Qualcomm Snapdragon chipset and likely have the same software (I would assume)

[pixel5.log](https://github.com/user-attachments/files/18078649/pixel5.log)

**Comment by caseyonline** [comment]
I updated a previous SIM (ever so slightly) that was exhibiting the same behavior by clearing the SMSP field.   Now, i get an
 _**Authentication Failure (MAC failure)**_ 
 reply from the UE.  

![Screenshot from 2024-12-17 18-10-21](https://github.com/user-attachments/assets/77650acd-62b5-4502-b449-4fa02b0ede32)

my authentication algorithm, UE key and OP/OPc appear to match, so I'm not sure why it's complaining  


_user_db.csv_,
`                                           ue1,mil,999690000000001,3c794f33e7cce49f73307bb6d0112a92,opc,17d131c47fdcfe394539b720e4759ee6,8000,0000000070ba,9,dynamic
`

![Screenshot1a](https://github.com/user-attachments/assets/e3a1af9d-4783-494d-9654-665fd4a8668f)

[enb_mac.pcap.txt](https://github.com/user-attachments/files/18173912/enb_mac.pcap.txt)

**Reply by andrepuschmann** [reply]
I am not familiar with that tool. Why don't you use pySim for that? There are many tutorials that show how to program your sim card correctly, e.g. https://docs.srsran.com/projects/project/en/latest/tutorials/source/cotsUE/source/index.html

I am going to close the issue until further notice .

**Reply by andrepuschmann** [reply]
Apologies for closing this @caseyonline. I was under the impression that it was under https://github.com/srsran/srsRAN_4G/issues.  Under issues we only collect actual code issues related to srsRAN.
When you opened this post I was initially worried that it might be a regression in the code or an otherwise undetected bug since you've tested with multiple phones already. However, since in your last post you've made progress after just touching the SIM card it's clear that it had nothing to do with our eNB. It looks like it's a SIM/UE authentication thing rather. I know for you the issue is still not solved but I am confident that by using our provided documentation and the tools mentioned in those you kind sort it out - as it worked for many others before. But go ahead and post your questions and report your steps here.

**Comment by caseyonline** [comment]
I was able to figure this out and got my UE to authenticate.  In short, the software used to program the SIM needs to be compatible with certain types of SIM cards.   pySim doesn't work with all cards including the ones I purchased.   As for my original issue when not seeing an Auth response, I can only suspect that a crash occurred during the reading and/or calculation because of the way I programmed the SIM.  

Big shout out to Phil Greenland for this insight and help.
https://www.quantulum.co.uk/blog/private-lte-with-limesdr-and-srsran--part-3-sim-cards/

**Comment by andrepuschmann** [comment]
Thanks for your feedback on this @caseyonline
